scripts/lint-package-jsons.ts

```diff
@@ -0,0 +1,68 @@
+import { execSync } from 'child_process';
+import { readFileSync } from 'fs';
+
+interface PackageJson {
+  private: boolean;
+  author: string;
+  engines: {
+    node: string;
+    pnpm: string;
+  };
+  packageManager: string;
+}
+
+const {
+  author: rootAuthor,
+  engines: { node: rootNodeVersion, pnpm: rootPnpmVersion },
+  packageManager: rootPackageManager,
+} = JSON.parse(readFileSync('package.json', 'utf-8')) as PackageJson;
+
+if (rootAuthor !== 'Fuel Labs <contact@fuel.sh> (https://fuel.network/)')
+  throw new Error(
+    "The package.json's author field should be 'Fuel Labs <contact@fuel.sh> (https://fuel.network/)' "
+  );
+
+const faultyPackageJsons = execSync('find packages/ apps/ -name package.json')
+  .toString()
+  .split('\n')
+  .filter(
+    (path) =>
+      path !== '' &&
+      !path.includes('/forc/') &&
+      !path.includes('/fuel-core/') &&
+      !path.match(/\/\..*\//) // internal folders like .next/package.json, .vitepress/package.json
+  )
```

Comment:
> [!WARNING]
> Nits-only. Feel free to ignore it.

```suggestion
const faultyPackageJsons = globSync(`{packages,apps}/*/package.json`)
  .filter((p) => !/\/forc|fuel-core\//.test(p))
```

---

If you fancy code/output aesthetics:

<details>
<summary>See code</summary>

```ts
import { error } from 'console';
import { readFileSync } from 'fs';
import { globSync } from 'glob';

interface PackageJson {
  private: boolean;
  author: string;
  engines: {
    node: string;
    pnpm: string;
  };
  packageManager: string;
}

const {
  author,
  engines: { node, pnpm },
  packageManager,
} = JSON.parse(readFileSync('package.json', 'utf-8')) as PackageJson;

const expectedAuthor = 'Fuel Labs <contact@fuel.sh> (https://fuel.network/)';

if (author !== expectedAuthor) {
  error(`The package.json's author field should be '${expectedAuthor}'`);
  process.exit(1);
}

const faultyPackageJsons = globSync(`{packages,apps}/*/package.json`)
  .filter((path) => !/\/forc|fuel-core\//.test(path))
  .filter((path) => {
    const json = JSON.parse(readFileSync(path, 'utf-8')) as PackageJson;
    return json.private
      ? false
      : json.author !== author ||
          json.engines.node !== node ||
          json.packageManager !== packageManager ||
          json.engines.pnpm !== pnpm;
  });

if (faultyPackageJsons.length) {
  const expectedConfigs = {
    author,
    engines: { node, pnpm },
    packageManager,
  };

  error(
    [
      `Expected configs:`,
      JSON.stringify(expectedConfigs, null, 2),
      `\nPlease review and correct these files:`,
      faultyPackageJsons.map((f) => ` - ${f}`).join('\n'),
    ].join('\n')
  );

  process.exit(1);
}
```
</details>

![errors](https://github.com/FuelLabs/fuels-ts/assets/26660/51afa690-7c58-4417-b2c1-f05b35a172d8)

Reply:
I fancy your suggestion!